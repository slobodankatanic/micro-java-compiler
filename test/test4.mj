//Test 302

program test302

const int jedan = 1;

int i;
bool ibool;

int niz2;

record Test {
 int niz[];
 int a;
}

Test t1, t2;

{
	
	int sum(int z, int d)
	{
		return z + d;
	}
	
	int inkrement(int a, int inc) 
		int pom; 
	{		
		pom = a;		
		return sum(a,pom);
	}
	
	bool verify() {
		print('v');
		return false;
	}	
	
	void main()
		int j; int cnd;
	{	
		 	
		i = 0;
	
		do {
			j = 0;
			
			do {
				cnd = 2;
				
				do {
					print('a');
					print('l');
					print('a');
					print(eol);
					cnd--;
				} while(cnd > 0);
				
				print(i, 1);
				print(',');
				print(j, 1);
				print(eol);
				j++;		
			} while(j < 3);
			
			i++;
			
			print(eol);
		} while(i < 5);
	}
}

