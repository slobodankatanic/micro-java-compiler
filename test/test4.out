ala
ala
0,0
ala
ala
0,1
ala
ala
0,2

ala
ala
1,0
ala
ala
1,1
ala
ala
1,2

ala
ala
2,0
ala
ala
2,1
ala
ala
2,2

ala
ala
3,0
ala
ala
3,1
ala
ala
3,2

ala
ala
4,0
ala
ala
4,1
ala
ala
4,2


Completion took 6 ms
